osgp/protocol-adapter-oslp/web-device-simulator/src/main/java/org/opensmartgridplatform/webdevicesimulator/domain/entities/Device.java

```diff
@@ -35,6 +36,7 @@
  * An entity class which contains the information of a single device.
  */
 @Entity
+@Proxy(lazy = false)
```

Comment:
The best solution would be DTO projection. 

Since the web-device-simulator project does not map entities to value objects or DTOs, the entities are used when rendering the JSP. Hibernate 5 lazy loads all properties of an entity, except when specified otherwise (for example by using `@Proxy(lazy = false)`. This means that every read operation of a property must be executed in a transaction. This behavior leaves the web-device-simulator application completely broken, as virtually every JSP render operation fails with null pointer exceptions or no session exceptions in particular.

Using a session or transaction when rendering JSPs (views) is an anti-pattern and has to be avoided at all times.